sdk/helper/custommetadata/custom_metadata.go

```diff
@@ -0,0 +1,66 @@
+package custommetadata
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-secure-stdlib/strutil"
+)
+
+type CustomMetadata map[string]string
+
+const (
+	maxKeys               = 64
+	maxKeyLength          = 128
+	maxValueLength        = 512
+	validationErrorPrefix = "custom_metadata validation failed"
+)
+
+func Validate(cm CustomMetadata) error {
```

Comment:
Should we add some comments to globally-scoped stuff, so people know what it is/how to use it? Or are these straightforward enough that no extra context is needed?

Reply:
Very thorough, thanks!